package com.ecommerce.product.repositories;

import aj.org.objectweb.asm.commons.Remapper;
import com.ecommerce.product.models.Product;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ProductRepository extends JpaRepository<Product, Long> {

    @Query("SELECT p FROM Product p WHERE p.active = true AND LOWER(p.name) LIKE LOWER(CONCAT('%', :keyword, '%')) AND p.stockQuantity > 0")
    List<Product> searchProduct(String keyword);

    Optional<Product> findByIdAndActiveTrue(Long id);
}
